relay-server/src/services/spooler/envelope_stack/sqlite.rs

```diff
@@ -484,36 +489,34 @@ mod tests {
             assert!(stack.push(envelope).await.is_ok());
         }
 
-        // We push 1 more envelope which results in spooling, which fails because of a database
-        // problem.
-        let envelope = mock_envelope(Instant::now());
-        assert!(matches!(
-            stack.push(envelope).await,
-            Err(SQLiteEnvelopeStackError::DatabaseError(_))
-        ));
+        // We push 1 more envelope that results in spooling, which returns ok but will fail in the
+        // async task that writes to the db.
+        let envelope_1 = mock_envelope(Instant::now());
+        assert!(stack.push(envelope_1.clone()).await.is_ok());
 
-        // The stack now contains the last of the 3 elements that were added. If we add a new one
-        // we will end up with 2.
-        let envelope = mock_envelope(Instant::now());
-        assert!(stack.push(envelope.clone()).await.is_ok());
-        assert_eq!(stack.batches_buffer_size, 3);
+        // We wait for the async spooling to take place.
+        sleep(Duration::from_millis(100)).await;
```

Comment:
You could make `push` return the join handle from `tokio::spawn`, then you can await it here.